src/snmp.c

```diff
@@ -1327,22 +1327,24 @@ static int csnmp_read_table(host_definition_t *host, data_definition_t *data) {
 
   status = 0;
   while (status == 0) {
-    int oid_list_todo_num;
-
     req = snmp_pdu_create(SNMP_MSG_GETNEXT);
     if (req == NULL) {
       ERROR("snmp plugin: snmp_pdu_create failed.");
       status = -1;
       break;
     }
 
-    oid_list_todo_num = 0;
+    size_t oid_list_todo_num = 0;
+    size_t var_idx[oid_list_len];
+    memset(&var_idx, 0, oid_list_len * sizeof(var_idx));
```

Comment:
This writes past the end of `var_idx` and will corrupt the stack. What you need is:

```c
memset(var_idx, 0, sizeof(var_idx));
```

Sorry I wasn't clear enough about that.

Reply:
Sorry, I'm not a qualified-enough C-programmer.)

Thanks for catching this.

Reply:
I disagree, you're obviously plenty qualified ;) Thanks for fixing :)

Reply:
I'm glad to co-work on this project! ;-)

What will the final decision on this patch?

Reply:
src/tcpconns.c:376

```
memset(&count_total, '\0', sizeof(count_total));
```
The declaration is:
```
static uint32_t count_total[TCP_STATE_MAX + 1];
```
)